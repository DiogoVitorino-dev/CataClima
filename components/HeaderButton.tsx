import { MaterialIcons } from "@expo/vector-icons";
import { Link } from "expo-router";
import { Pressable, StyleProp, TextStyle } from "react-native";

export default function HeaderButton ({href,params, icon,iconSize, iconColor, style} : 
    {href:string,params?:any, iconSize:number, icon:any, iconColor:string , style?:StyleProp<TextStyle>}) {

    return (
        <Link href={{pathname:href, params:params}} style={style} asChild>
            <Pressable>
            {({ pressed }) => (
                <MaterialIcons
                name={icon}
                size={iconSize}
                color={iconColor}
                style={{opacity: pressed ? 0.5 : 1 }}
                />
            )}
            </Pressable>
        </Link>
    );
}