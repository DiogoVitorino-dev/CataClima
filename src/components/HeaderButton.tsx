import {MaterialIcons} from '@expo/vector-icons';
import {Link} from 'expo-router';
import {Pressable, StyleProp, TextStyle} from 'react-native';

export default function HeaderButton({
  href,
  params,
  onPress,
  icon,
  iconSize,
  iconColor,
  style,
}: {
  href?: string;
  params?: object;
  onPress?:() => void;
  iconSize: number;
  icon: string | any;
  iconColor: string;
  style?: StyleProp<TextStyle>;
}) {
  return (
    <Link 
      href={{pathname: href, params: params}} 
      style={style} 
      asChild
    >
      <Pressable 
        accessibilityRole='button'
        onPress={() => onPress ? onPress() : null}>
        {({pressed}) => (
          <MaterialIcons 
            name={icon} 
            size={iconSize} 
            color={iconColor} 
            style={{opacity: pressed ? 0.5 : 1}} 
          />
        )}
      </Pressable>
    </Link>
  );
}
