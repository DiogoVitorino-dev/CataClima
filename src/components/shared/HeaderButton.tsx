import { FontAwesome } from "@expo/vector-icons";
import { Pressable, StyleProp, TextStyle } from "react-native";

import { IconsFontAwesome } from "@/constants/TypesApp";

interface IProps {
  onPress: () => void;
  icon: IconsFontAwesome;
  color?: string;
  style?: StyleProp<TextStyle>;
}

export default function HeaderButton({ onPress, icon, color, style }: IProps) {
  return (
    <Pressable accessibilityRole="button" style={style} onPress={onPress}>
      {({ pressed }) => (
        <FontAwesome
          name={icon}
          size={30}
          color={color}
          style={{ opacity: pressed ? 0.5 : 1 }}
        />
      )}
    </Pressable>
  );
}
