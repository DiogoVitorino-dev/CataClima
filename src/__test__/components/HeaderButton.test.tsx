import {describe,it, expect, afterEach} from '@jest/globals';
import { cleanup } from '@testing-library/react-native';
import renderer from 'react-test-renderer'
import HeaderButton from '../../components/HeaderButton';

describe('HeaderButton component', () => {
  afterEach(cleanup)

  it('should renders correctly', () => {    
      const tree = renderer.create(
      <HeaderButton
        href='/'
        onPress={()=>{}}
        params={{test:'test'}}
        style={{margin:5}}
        icon='cloud' 
        iconColor='#fff' 
        iconSize={20} 
      />).toJSON();
      
      expect(tree).toMatchSnapshot()    
  }) 
})