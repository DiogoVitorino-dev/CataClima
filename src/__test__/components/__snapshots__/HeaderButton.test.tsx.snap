// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderButton component should renders correctly 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  href="/?test=test"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "margin": 5,
    }
  }
>
  <Text />
</View>
`;
